src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -22,58 +21,172 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
         public Task<IEnumerable<KeyValuePair<string, string>>> ProcessKeyValue(ConfigurationSetting setting, Logger logger, CancellationToken cancellationToken)
         {
-            FeatureFlag featureFlag;
-            try
-            {
-                 featureFlag = JsonSerializer.Deserialize<FeatureFlag>(setting.Value);
-            }
-            catch (JsonException e)
-            {
-                throw new FormatException(setting.Key, e);
-            }
-
             var keyValues = new List<KeyValuePair<string, string>>();
 
-            if (featureFlag.Enabled)
+            try
```

Comment:
Mixing up the json deserialization with the configuration authoring is hard to follow.

I'd like to see a flow where we go from JSON to a feature flag object, and then use the feature flag object to author the configuration.

Reply:
I was trying to avoid the extra steps needed, but I can see how it makes this easier to follow after it got more complex than I expected. I can just add back the files I deleted that model the feature flag.